### DISCUSSION TITLE
Perhaps let parameter be constrained to another function's parameter

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1171] [General]
let ``` type 'a ``` be ``` f (x:'a) ``` we can have ``` f1 x =  'b() |> (f2 (p'a) (q'b)) ```
It helps to partially apply functions that we can't today
![image](https://user-images.githubusercontent.com/35312807/185593880-79a328f0-2bcd-4f3f-af4d-db0fe503cd1e.png)

**Comment by vzarytovskii** [comment]
If you'd like to make a suggestion, please turn it into an issue and use the template from this repository (automaticky fills in when opening the issue).
